src/core/hle/service/am/service/application_functions.cpp

```diff
@@ -267,6 +268,25 @@ Result IApplicationFunctions::GetSaveDataSizeMax(Out<u64> out_max_normal_size,
     R_SUCCEED();
 }
 
+Result IApplicationFunctions::GetCacheStorageMax(Out<u32> out_cache_storage_index_max,
+                                                 Out<u64> out_max_journal_size) {
+    LOG_DEBUG(Service_AM, "called");
+
+    const auto title_id = m_applet->program_id;
+
+    std::vector<u8> nacp;
+    const auto result = system.GetARPManager().GetControlProperty(&nacp, title_id);
+
+    if (R_SUCCEEDED(result)) {
+        const auto rawnacp = reinterpret_cast<FileSys::RawNACP*>(nacp.data());
+
+        *out_cache_storage_index_max = static_cast<u32>(rawnacp->cache_storage_max_index);
+        *out_max_journal_size = static_cast<u64>(rawnacp->cache_storage_data_and_journal_max_size);
+    }
+
+     R_SUCCEED();
```

Comment:
Take advantage of the behavior of R_TRY, and don't alias a u8* with a RawNACP*:
```suggestion
    FileSys::RawNACP raw_nacp{};
    std::vector<u8> nacp;
    R_TRY(system.GetARPManager().GetControlProperty(&nacp, m_applet->program_id));

    std::memcpy(&raw_nacp, nacp.data(), std::min(sizeof(raw_nacp), nacp.size()));

    *out_cache_storage_index_max = static_cast<u32>(raw_nacp->cache_storage_max_index);
    *out_max_journal_size = static_cast<u64>(raw_nacp->cache_storage_data_and_journal_max_size);

    R_SUCCEED();
```

Also, please fix the nacp [field](https://github.com/yuzu-emu/yuzu/blob/master/src/core/file_sys/control_metadata.h#L67) for cache_storage_max_index ... it is a u16 now as some fields adjacent to it have been [split](https://switchbrew.org/wiki/NACP#Structure) to expand the structure.